### PR TITLE
[Documentation] Update documentation regarding ReCaptcha in graphQL and REST requests

### PR DESCRIPTION
**Pull request by vkolesny**
## Purpose of this pull request

This pull request (PR) is a continuation of the https://github.com/magento-commerce/devdocs/pull/3042 Pull Request.
We should provide more clarity that "X-Captcha" and "X-ReCaptcha" headers can be generated by web form widgets in the background only.
Please provide more clear explanation regarding X-Captcha and X-ReCaptcha headers. Please add to the documentation examples how this headers can be added to the requests.

## Affected pages

- https://developer.adobe.com/commerce/webapi/graphql/usage/protected-mutations/
- https://developer.adobe.com/commerce/webapi/rest/use-rest/protected-endpoints/

### CODE REVIEW

**Comment by keharper** [comment]
@vkolesny Are the changes in the PR due to changes in the 2.4.7 release, or are they describing released code?

**Comment by vkolesny** [comment]
Hi @keharper ,
These changes are for already existing code.

It is just to add more clarity and explanation to already existing feature.

File: src/pages/graphql/usage/protected-mutations.md [hunk 1]
```diff
@@ -8,7 +8,9 @@ keywords:
 
 # Protected mutations
 
-If CAPTCHA or reCAPTCHA is enabled on pages requiring shopper input, then in most cases, the corresponding mutations that send requests to the application server must include an HTTP header that contains a value entered by the shopper (for CAPTCHA) or generated by the Google API (for reCAPTCHA). However, if you specify an integration authorization token in the header of the mutation, then you do not supply a header specific to CAPTCHA or reCAPTCHA.
+If CAPTCHA or reCAPTCHA is enabled on pages requiring shopper input, then in most cases, the corresponding mutations that send requests to the application server must include an HTTP header that contains a value entered by the shopper (for CAPTCHA) or generated by the Google API (for reCAPTCHA). These headers are generated by UI Web form widgets during interacting with shopper. However, if you specify an integration authorization token in the header of the mutation, then you do not supply a header specific to CAPTCHA or reCAPTCHA.
```
Comment (keharper) [inline src/pages/graphql/usage/protected-mutations.md:11]:
```suggestion
If CAPTCHA or reCAPTCHA is enabled on pages requiring shopper input, then in most cases, the corresponding mutations that send requests to the application server must include an HTTP header that contains a value entered by the shopper (for CAPTCHA) or generated by the Google API (for reCAPTCHA). These headers are generated by UI Web form widgets during interactions with the shopper. However, if you specify an integration authorization token in the header of the mutation, then you do not supply a header specific to CAPTCHA or reCAPTCHA.
```

File: src/pages/graphql/usage/protected-mutations.md [hunk 2]
```diff
@@ -8,7 +8,9 @@ keywords:
...
+
+Even when Adobe Commerce is used in a headless way, like with PWA, the Merchant must have their own Web UI which must include rendering the correct Captcha or reCAPTCHA web form widget, to capture the proper `X-Captcha` or `X-ReCaptcha` HTTP header, and send it in the API request in the background only.
```
Comment (keharper) [inline src/pages/graphql/usage/protected-mutations.md:13]:
```suggestion
Even when Adobe Commerce is used in a headless way, like with PWA, the merchant must have their own Web UI that renders the correct Captcha or reCAPTCHA web form widget, captures the proper `X-Captcha` or `X-ReCaptcha` HTTP header, and sends it in the API request in the background only.
```

File: src/pages/rest/use-rest/protected-endpoints.md [hunk 3]
```diff
@@ -8,7 +8,9 @@ keywords:
  
 # Protected endpoints
 
-If CAPTCHA or reCAPTCHA is enabled on pages requiring shopper input, then in most cases, the corresponding endpoints that send requests to the Adobe Commerce server must include an HTTP header that contains a value entered by the shopper (for CAPTCHA) or generated by the Google API (for reCAPTCHA). However, if you specify an integration authorization token in the header of the endpoint, then you do not supply a header specific to CAPTCHA or reCAPTCHA.
+If CAPTCHA or reCAPTCHA is enabled on pages requiring shopper input, then in most cases, the corresponding endpoints that send requests to the Adobe Commerce server must include an HTTP header that contains a value entered by the shopper (for CAPTCHA) or generated by the Google API (for reCAPTCHA). These headers are generated by UI Web form widgets during interacting with shopper. However, if you specify an integration authorization token in the header of the endpoint, then you do not supply a header specific to CAPTCHA or reCAPTCHA.
```
Comment (keharper) [inline src/pages/rest/use-rest/protected-endpoints.md:11]:
```suggestion
If CAPTCHA or reCAPTCHA is enabled on pages requiring shopper input, then in most cases, the corresponding endpoints that send requests to the Adobe Commerce server must include an HTTP header that contains a value entered by the shopper (for CAPTCHA) or generated by the Google API (for reCAPTCHA). These headers are generated by UI Web form widgets during interactions with the shopper. However, if you specify an integration authorization token in the header of the endpoint, then you do not supply a header specific to CAPTCHA or reCAPTCHA.
```

File: src/pages/rest/use-rest/protected-endpoints.md [hunk 4]
```diff
@@ -8,7 +8,9 @@ keywords:
...
+
+Even when Adobe Commerce is used in a headless way, like with PWA, the Merchant must have their own Web UI which must include rendering the correct Captcha or reCAPTCHA web form widget, to capture the proper `X-Captcha` or `X-ReCaptcha` HTTP header, and send it in the API request in the background only.
```
Comment (keharper) [inline src/pages/rest/use-rest/protected-endpoints.md:13]:
```suggestion
Even when Adobe Commerce is used in a headless way, like with PWA, the merchant must have their own Web UI that renders the correct Captcha or reCAPTCHA web form widget, captures the proper `X-Captcha` or `X-ReCaptcha` HTTP header, and sends it in the API request in the background only.
```

**Review by keharper — CHANGES_REQUESTED** [review]

**Comment by keharper** [comment]
@vkolesny This PR is failing validation checks because external links in unrelated topics are suddenly breaking. I fixed those broken links in another PR. If you're using the GH web UI to incorporate my comments, click **Update branch**.  Otherwise, pull in the latest changes from `main`.

**Comment by vkolesny** [comment]
Thanks, @keharper.
Proposed changes were committed.
Latest main was pulled into PR.

**Review by keharper — APPROVED** [review]

**Pull request merged**